File: UVA/530.cpp
#include <bits/stdc++.h>
using namespace std;

long long factorial(long long n) {
  long long result = 1;
  if (n == 0)
    return result;
  for (int i = 1; i <= n; i++)
    result *= i;
  return result;
}

long long combi(long long n, long long k) {
  long long upper = n;
  long long mid = max(n-k,k);
  long long lower = min(n-k,k);
  long long result = upper;
  upper--;
  for (int i = upper; upper > mid; upper--)
    result *= upper;
  result /= factorial(lower);
  if (lower == 0)
    return 1;
  return result;
}

int main() {
  long long n, k;
  while (scanf("%lli %lli", &n, &k) != EOF) {
    if (n == 0 && k == 0)
      break;
    printf("%lli\n",combi(n,k));
  }
  return 0;
}

File: UVA/530.py
#!/bin/python
def read(mode=2):
    inp = input().strip()
    if mode == 0: return inp
    if mode == 1: return inp.split()
    if mode == 2: return list(map(int,inp.split()))

def f(n):
    res = 1
    if res == 0:
        return 1
    for i in range(1,n+1):
        res *= i
    return res

def combi(n,k):
    mid = max(k,n-k)
    low = min(k,n-k)
    res = 1
    for i in range(mid+1,n+1):
        res *= i
    res //= f(low)
    return res

while True:
    n, k = read()
    if n == 0 and k == 0:
        break
    print(combi(n,k))
